src/main/java/seedu/address/model/schedule/Event.java

```diff
@@ -53,31 +60,37 @@ public LocalTime getTime() {
         return time;
     }
 
-    public LocalTime getEndTime() {
-        return time.plus(duration);
+    public LocalDate getEndDate() {
+        return date.atTime(time).plus(duration).toLocalDate();
     }
 
-    public LocalDate getEndDate() {
-        LocalDateTime endDateTime = date.atTime(time).plus(duration);
-        return endDateTime.toLocalDate();
+    public LocalTime getEndTime() {
+        return time.plus(duration);
     }
 
     public Duration getDuration() {
         return duration;
     }
 
-    private String getDurationDays() {
-        long days = duration.toDaysPart();
-        if (days > 0) {
-            return String.format(" (+%s)", days);
-        }
-        return "";
-    }
-
     public EventDescription getEventDescription() {
         return eventDescription;
     }
 
+    public Optional<RecurFrequency> getRecurFrequency() {
+        return Optional.ofNullable(recurFrequency);
+    }
+
+    /**
+     * Returns true if the Event is recurring.
+     * @return
+     */
+    public boolean isRecurring() {
+        if (recurFrequency == null) {
```

Comment:
is getRecurFrequency().isPresent() better?

Reply:
Sounds good.